### PR TITLE
3.0 update: intro to RP

### LINKED ISSUE #12678: Improve CSAT of intro to RP
Update code to 3.0

image

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 164d7efe-01bf-0b26-7a23-f10704f338f5
* Version Independent ID: b3e2bc87-1582-2d34-c7c2-7c8e901ac771
* Content: [Introduction to Razor Pages in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/index?view=aspnetcore-2.0&branch=pr-en-us-12676&tabs=visual-studio)
* Content Source: [aspnetcore/razor-pages/index.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/razor-pages/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-razorpages**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### LINKED ISSUE #13377: 3.0 update: Intro to RP
RP

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 164d7efe-01bf-0b26-7a23-f10704f338f5
* Version Independent ID: b3e2bc87-1582-2d34-c7c2-7c8e901ac771
* Content: [Introduction to Razor Pages in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/index?view=aspnetcore-2.2&tabs=visual-studio)
* Content Source: [aspnetcore/razor-pages/index.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/razor-pages/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-razorpages**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #13377 Fixes  #12678

[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/razor-pages/index?view=aspnetcore-3.0&branch=pr-en-us-13993&tabs=visual-studio)

Requires reviewer checkoff

- [ ]  The configuration and settings in following sections is not required by most apps.
- [ ]  Review [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/razor-pages/index?view=aspnetcore-3.0&branch=pr-en-us-13993&tabs=visual-studio)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@rynowak  please review. You'll need to review the entire [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/razor-pages/index?view=aspnetcore-3.0&branch=pr-en-us-13993&tabs=visual-studio)

File: aspnetcore/razor-pages/index.md [hunk 1]
```diff
@@ -9,9 +9,11 @@ uid: razor-pages/index
 ---
 # Introduction to Razor Pages in ASP.NET Core
 
+::: moniker range=">= aspnetcore-3.0"
+
 By [Rick Anderson](https://twitter.com/RickAndMSFT) and [Ryan Nowak](https://github.com/rynowak)
 
-Razor Pages is a new aspect of ASP.NET Core MVC that makes coding page-focused scenarios easier and more productive.
+Razor Pages can make coding page-focused scenarios easier and more productive that using controllers and views.
```
Comment (rynowak) [inline aspnetcore/razor-pages/index.md]:
```suggestion
Razor Pages can make coding page-focused scenarios easier and more productive than using controllers and views.
```

File: aspnetcore/razor-pages/index.md [hunk 2]
```diff
@@ -41,35 +43,608 @@ This document provides an introduction to Razor Pages. It's not a step by step t
...
+
+  * *Index.cshtml*
+  * *Privacy.cshtml*
+  * */Customers*
+
+    * *Create.cshtml*
+    * *Edit.cshtml*
+    * *Index.cshtml*
+
+The *Pages/Customers/Create.cshtml* and *Pages/Customers/Edit.cshtml* pages redirect to *Pages/Customers/Index.cshtml* after success. The string `./Index` is part of the URI to access the preceding page. The string `./Index` can be used to generate URIs to the *Pages/Customers/Index.cshtml* page. For example:
```
Comment (rynowak) [inline aspnetcore/razor-pages/index.md]: > The string `./Index` is part of the URI to access the preceding page.

I  think it's really really really important that we don't refer to the strings passed into link generation **as URIs**. This has been a point of confusion for users and I want to make sure we're not giving people the wrong concept.

The correct terminology (page name) is introduced a few paragraphs down and has a good description.

Comment (Rick-Anderson) [reply aspnetcore/razor-pages/index.md]: @rynowak can you suggest better wording?

Comment (Rick-Anderson) [reply aspnetcore/razor-pages/index.md]: @scottaddie please suggest better wording.

**Pull request merged**